src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -173,6 +202,898 @@ - (void)subscribeWithQueue:(dispatch_queue_t)queue
+                        [[NSDictionary alloc] initWithObjectsAndKeys:[NSNumber numberWithUnsignedInt:attribPath.mEndpointId],
+                                              kCHIPEndpointIdKey, [NSNumber numberWithUnsignedInt:attribPath.mClusterId],
+                                              kCHIPClusterIdKey, [NSNumber numberWithUnsignedInt:attribPath.mAttributeId],
+                                              kCHIPAttributeIdKey, data.GetDecodedObject(), kCHIPValueKey, nil]);
+                }
+            };
+
+            auto establishedOrFailed = std::make_shared<BOOL>(NO);
+            auto onFailureCb
+                = [context, failureCb, establishedOrFailed](const app::ConcreteAttributePath * attribPath, CHIP_ERROR error) {
+                      if (!(*establishedOrFailed)) {
+                          *establishedOrFailed = YES;
+                          NSObjectAttributeCallbackSubscribeBridge::OnSubscriptionEstablished(context);
+                      }
+                      if (failureCb != nullptr) {
+                          failureCb(context, error);
+                      }
+                  };
+
+            auto onEstablishedCb = [context, establishedOrFailed]() {
+                if (*establishedOrFailed) {
+                    return;
+                }
+                *establishedOrFailed = YES;
+                NSObjectAttributeCallbackSubscribeBridge::OnSubscriptionEstablished(context);
+            };
+
+            auto chipEndpointId = static_cast<chip::EndpointId>(endpointId);
+            auto chipClusterId = static_cast<chip::ClusterId>(clusterId);
+            auto chipAttributeId = static_cast<chip::AttributeId>(attributeId);
+
+            app::AttributePathParams attributePath(chipEndpointId, chipClusterId, chipAttributeId);
+            app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
+            CHIP_ERROR err = CHIP_NO_ERROR;
+
+            chip::app::ReadPrepareParams readParams([self internalDevice]->GetSecureSession().Value());
+            readParams.mpAttributePathParamsList = &attributePath;
+            readParams.mAttributePathParamsListSize = 1;
+
+            auto onDone = [](BufferedReadAttributeCallback<NSObjectData> * callback) { chip::Platform::Delete(callback); };
+
+            auto callback = chip::Platform::MakeUnique<BufferedReadAttributeCallback<NSObjectData>>(
+                chipClusterId, chipAttributeId, onReportCb, onFailureCb, onDone, onEstablishedCb);
+            VerifyOrReturnError(callback != nullptr, CHIP_ERROR_NO_MEMORY);
+
+            auto readClient = chip::Platform::MakeUnique<app::ReadClient>(engine, [self internalDevice]->GetExchangeManager(),
+                callback -> GetBufferedCallback(), chip::app::ReadClient::InteractionType::Subscribe);
+            VerifyOrReturnError(readClient != nullptr, CHIP_ERROR_NO_MEMORY);
+
+            err = readClient->SendAutoResubscribeRequest(std::move(readParams));
```

Comment:
This is going to crash, no?  The callback will try to delete the paths when `OnDeallocatePaths` happens, but the paths are stack-allocated, and things will go sideways.

Reply:
You probably looked at an older commit. Paths are heap allocated now.

Reply:
I was looking specifically at this PR, not the tree as it is on tip so far.  If this is fixed, great.

Reply:
I just looked on tip, and there the allocation/deallocation uses different allocators.  The allocation uses `new` and the deallocation uses `Platform::Delete`.   Those might use the same heap, or they might not, depending.  We should make them match by either using `Platform::New` or by using `delete`.

Reply:
Should use `Platform::New` then.